control-plane/connect-inject/webhook_v2/consul_dataplane_sidecar.go

```diff
@@ -0,0 +1,474 @@
+		container.Env = append(container.Env, corev1.EnvVar{
+			Name: "DP_ENVOY_READY_BIND_ADDRESS",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{FieldPath: "status.podIP"},
+			},
+		})
+		// Configure the port on which the readiness probe will query the proxy for its health.
+		container.Ports = append(container.Ports, corev1.ContainerPort{
+			Name:          fmt.Sprintf("%s", "proxy-health"),
+			ContainerPort: int32(constants.ProxyDefaultHealthPort),
+		})
+	}
+
+	// Add any extra VolumeMounts.
+	if userVolMount, ok := pod.Annotations[constants.AnnotationConsulSidecarUserVolumeMount]; ok {
+		var volumeMounts []corev1.VolumeMount
+		err := json.Unmarshal([]byte(userVolMount), &volumeMounts)
+		if err != nil {
+			return corev1.Container{}, err
+		}
+		container.VolumeMounts = append(container.VolumeMounts, volumeMounts...)
+	}
+
+	tproxyEnabled, err := common.TransparentProxyEnabled(namespace, pod, w.EnableTransparentProxy)
+	if err != nil {
+		return corev1.Container{}, err
+	}
+
+	// If not running in transparent proxy mode and in an OpenShift environment,
+	// skip setting the security context and let OpenShift set it for us.
+	// When transparent proxy is enabled, then consul-dataplane needs to run as our specific user
+	// so that traffic redirection will work.
+	if tproxyEnabled || !w.EnableOpenShift {
+		if pod.Spec.SecurityContext != nil {
+			// User container and consul-dataplane container cannot have the same UID.
+			if pod.Spec.SecurityContext.RunAsUser != nil && *pod.Spec.SecurityContext.RunAsUser == sidecarUserAndGroupID {
+				return corev1.Container{}, fmt.Errorf("pod's security context cannot have the same UID as consul-dataplane: %v", sidecarUserAndGroupID)
+			}
+		}
+		// Ensure that none of the user's containers have the same UID as consul-dataplane. At this point in injection the meshWebhook
+		// has only injected init containers so all containers defined in pod.Spec.Containers are from the user.
+		for _, c := range pod.Spec.Containers {
+			// User container and consul-dataplane container cannot have the same UID.
+			if c.SecurityContext != nil && c.SecurityContext.RunAsUser != nil && *c.SecurityContext.RunAsUser == sidecarUserAndGroupID && c.Image != w.ImageConsulDataplane {
+				return corev1.Container{}, fmt.Errorf("container %q has runAsUser set to the same UID \"%d\" as consul-dataplane which is not allowed", c.Name, sidecarUserAndGroupID)
+			}
+		}
+		container.SecurityContext = &corev1.SecurityContext{
+			RunAsUser:              pointer.Int64(sidecarUserAndGroupID),
+			RunAsGroup:             pointer.Int64(sidecarUserAndGroupID),
+			RunAsNonRoot:           pointer.Bool(true),
+			ReadOnlyRootFilesystem: pointer.Bool(true),
+		}
+	}
+
+	return container, nil
+}
+
+func (w *MeshWebhook) getContainerSidecarArgs(namespace corev1.Namespace, bearerTokenFile string, pod corev1.Pod) ([]string, error) {
+	proxyIDFileName := "/consul/mesh-inject/proxyid"
```

Comment:
I wasn't planning on using the proxyIDFile in `connect-init`/`mesh-init` anymore since we can just use the Pod name. I think this means we'll need to remove this arg and add a new environment variable `DP_PROXY_ID` that points to the pod name: https://github.com/hashicorp/consul-dataplane/pull/242/files#diff-36c1ea42620072b194bdd2aebc8225b33766ae2b2fc526d0f2bcd2c1640fd093R65-R67